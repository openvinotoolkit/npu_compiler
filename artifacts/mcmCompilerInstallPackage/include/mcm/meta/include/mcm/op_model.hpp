/*
    DO NOT MODIFY - that file was generated automatically using op::OpRegistry::generateCompositionAPI()
*/

#ifndef MV_OP_MODEL_HPP_
#define MV_OP_MODEL_HPP_

#include "include/mcm/compositional_model.hpp"
#include "include/mcm/computation/model/base_op_model.hpp"

#include "include/mcm/compiler/compilation_profiler.hpp"

namespace mv

{

    class OpModel: public BaseOpModel, public CompositionalModel
    {

    private:
    public:

        OpModel(const std::string& name);
        OpModel(ComputationModel& model);
        virtual ~OpModel();

        mv::Data::TensorIterator align(Data::TensorIterator data, const std::size_t& dimension = 2, const std::size_t& pad = 16, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator argmax(Data::TensorIterator data, const int64_t& out_max_val, const int64_t& top_k, const int64_t& axis = 99, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator averagePool(Data::TensorIterator data, const std::array<unsigned short, 2>& kSize, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const bool& exclude_pad = true, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator barrierTask(const Barrier& Barrier, const std::string& name = "");
        mv::Data::TensorIterator batchNormalization(Data::TensorIterator data, Data::TensorIterator mean, Data::TensorIterator variance, Data::TensorIterator offset, Data::TensorIterator scale, const double& eps, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator bias(Data::TensorIterator data, Data::TensorIterator weights, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator concat(const std::vector< Data::TensorIterator >& inputs, const std::string& axis = "C", const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator constant(const std::vector<double>& data, const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator constantDataElement(const std::vector<mv::DataElement>& data, const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator constantInt(const std::vector<int64_t>& data, const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator conv(Data::TensorIterator data, Data::TensorIterator weights, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const unsigned& dilationFactor = 1, const unsigned& group = 1, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator conversion(Data::TensorIterator data, const Order& order, const std::string& name = "");
        mv::Data::TensorIterator copy(Data::TensorIterator data, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator crop(Data::TensorIterator data, const std::size_t& cropVal, const std::size_t& dimension = 2, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator custom(const std::vector< Data::TensorIterator >& inputs, const std::vector<uint8_t>& kernelData, const std::vector<uint8_t>& paramData, const Order& outOrder, const Shape& outShape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator dMATask(Data::TensorIterator data, const DmaDirection& direction, const std::string& name = "");
        mv::Data::TensorIterator dPUTaskConv(const std::vector< Data::TensorIterator >& inputs, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const unsigned& dilationFactor = 1, const unsigned& group = 1, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator dPUTaskMaxPool(const std::vector< Data::TensorIterator >& inputs, const std::array<unsigned short, 2>& kSize, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const bool& exclude_pad = true, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator dPUTaskDepthwiseConv(const std::vector< Data::TensorIterator >& inputs, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const unsigned& dilationFactor = 1, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator dPUTaskEltwise(const std::vector< Data::TensorIterator >& inputs, const std::string& eltwiseType, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator deallocate(Data::TensorIterator inputs, const std::string& name = "");
        mv::Data::TensorIterator depthwiseConv(Data::TensorIterator data, Data::TensorIterator weights, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const unsigned& dilationFactor = 1, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator detectionOutput(const std::vector< Data::TensorIterator >& inputs, const int64_t& num_classes, const int64_t& keep_top_k, const double& nms_threshold, const int64_t& background_label_id, const int64_t& top_k, const bool& variance_encoded_in_target, const std::string& code_type, const bool& share_location, const double& confidence_threshold, const bool& clip_before_nms, const bool& clip_after_nms, const int64_t& decrease_label_id, const bool& normalized, const int64_t& input_height, const int64_t& input_width, const double& objectness_score, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator dropout(Data::TensorIterator input, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator dummy(Data::TensorIterator data, const std::string& name = "") override;
        mv::Data::TensorIterator eltwise(const std::vector< Data::TensorIterator >& inputs, const std::string& eltwiseType, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator elu(Data::TensorIterator data, const unsigned& alpha = 1, const std::string& name = "") override;
        mv::Data::TensorIterator fakeQuantize(Data::TensorIterator data, Data::TensorIterator input_min, Data::TensorIterator input_max, Data::TensorIterator output_min, Data::TensorIterator output_max, const unsigned& levels, const std::string& name = "") override;
        mv::Data::TensorIterator flatten(Data::TensorIterator input, const int64_t& axis = 1, const int64_t& end_axis = 3, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator fullyConnected(Data::TensorIterator data, Data::TensorIterator weights, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator identity(Data::TensorIterator data, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator implicitConcat(const std::vector< Data::TensorIterator >& inputs, const std::string& axis = "C", const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator implicitInput(Data::TensorIterator data, const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator implicitInputSlice(Data::TensorIterator inputs, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator implicitOutput(Data::TensorIterator data, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator implicitPermute(Data::TensorIterator inputs, const Shape& shape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator implicitReshape(Data::TensorIterator inputs, const Shape& shape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator implicitUnion(const std::vector< Data::TensorIterator >& inputs, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator input(const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const bool& networkInput = true, const std::string& name = "") override;
        mv::Data::TensorIterator interp(Data::TensorIterator data, const double& factor, const unsigned& pad_beg, const unsigned& pad_end, const unsigned& height = 0, const unsigned& width = 0, const bool& align_corners = true, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator leakyRelu(Data::TensorIterator data, const double& alpha = 0, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator localResponseNormalization(Data::TensorIterator data, const unsigned& size, const unsigned& bias, const std::string& name = "") override;
        mv::Data::TensorIterator matMul(Data::TensorIterator data0, Data::TensorIterator data1, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator maxPool(Data::TensorIterator data, const std::array<unsigned short, 2>& kSize, const std::array<unsigned short, 2>& stride, const std::array<unsigned short, 4>& padding, const bool& exclude_pad = true, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator maximum(Data::TensorIterator inputs, const double& maximum, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator minimum(Data::TensorIterator inputs, const double& minimum, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator norm(Data::TensorIterator data, const double& alpha, const double& beta, const std::string& region, const unsigned& local_size, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator normalize(Data::TensorIterator data, Data::TensorIterator weights, const double& eps, const unsigned& across_spatial = 0, const unsigned& channel_shared = 0, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator output(Data::TensorIterator data, const DType& precision = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const bool& networkOutput = true, const std::string& name = "") override;
        mv::Data::TensorIterator pSROIPooling(const std::vector< Data::TensorIterator >& inputs, const std::size_t& output_dim, const std::size_t& group_size, const double& spatial_scale, const std::size_t& pooled_w, const std::size_t& pooled_h, const std::size_t& spatial_bin_x, const std::size_t& spatial_bin_y, const std::string& mode, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator permute(Data::TensorIterator data, const Order& order, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator placeholderTask(const Shape& shape, const DType& dType, const Order& order, const std::string& name = "");
        mv::Data::TensorIterator prelu(Data::TensorIterator data, Data::TensorIterator slope, const std::string& name = "") override;
        mv::Data::TensorIterator priorbox(const std::vector< Data::TensorIterator >& inputs, const unsigned& flip, const unsigned& clip, const double& step_w, const double& step_h, const double& offset, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator proposal(const std::vector< Data::TensorIterator >& inputs, const unsigned& base_size, const unsigned& pre_nms_topn, const unsigned& post_nms_topn, const double& nms_thresh, const unsigned& feat_stride, const unsigned& min_size, const double& pre_nms_thresh = 0.000000, const bool& clip_before_nms = true, const bool& clip_after_nms = false, const bool& normalize = false, const double& box_size_scale = 1.000000, const double& box_coordinate_scale = 1.000000, const std::string& framework = "TENSORFLOW", const bool& for_deformable = false, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator quantize(Data::TensorIterator data, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator rOIPooling(const std::vector< Data::TensorIterator >& inputs, const unsigned& pooled_w, const unsigned& pooled_h, const double& spatial_scale, const unsigned& roi_pooling_method, const unsigned& num_rois, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator regionYolo(Data::TensorIterator data, const unsigned& coords, const unsigned& classes, const bool& do_softmax, const unsigned& num = 0, const std::vector<unsigned>& mask = {}, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator relu(Data::TensorIterator data, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator reorder(Data::TensorIterator data, const Order& order, const std::string& name = "") override;
        mv::Data::TensorIterator reorgYolo(Data::TensorIterator data, const unsigned& stride, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator resample(Data::TensorIterator input, const std::string& interpolation, const bool& antialias, const Shape& output_shape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator reshape(Data::TensorIterator data, const Shape& shape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator scale(Data::TensorIterator data, Data::TensorIterator weights, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator sigmoid(Data::TensorIterator data, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator slice(Data::TensorIterator data, const Shape& begin, const Shape& size, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator softmax(Data::TensorIterator data, const std::string& axis = "C", const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator sparsityMap(const std::vector<int64_t>& data, const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator tanh(Data::TensorIterator data, const std::string& name = "") override;
        mv::Data::TensorIterator topK(Data::TensorIterator data, const std::string& sort, const std::string& mode, const int64_t& top_k, const int64_t& axis = 99, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "") override;
        mv::Data::TensorIterator uPATaskDummy(const std::vector< Data::TensorIterator >& inputs, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskIdentity(const std::vector< Data::TensorIterator >& inputs, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskSoftmax(const std::vector< Data::TensorIterator >& inputs, const std::string& axis = "C", const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskProposal(const std::vector< Data::TensorIterator >& inputs, const unsigned& base_size, const unsigned& pre_nms_topn, const unsigned& post_nms_topn, const double& nms_thresh, const unsigned& feat_stride, const unsigned& min_size, const double& pre_nms_thresh = 0.000000, const bool& clip_before_nms = true, const bool& clip_after_nms = false, const bool& normalize = false, const double& box_size_scale = 1.000000, const double& box_coordinate_scale = 1.000000, const std::string& framework = "TENSORFLOW", const bool& for_deformable = false, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskROIPooling(const std::vector< Data::TensorIterator >& inputs, const unsigned& pooled_w, const unsigned& pooled_h, const double& spatial_scale, const unsigned& roi_pooling_method, const unsigned& num_rois, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskPSROIPooling(const std::vector< Data::TensorIterator >& inputs, const std::size_t& output_dim, const std::size_t& group_size, const double& spatial_scale, const std::size_t& pooled_w, const std::size_t& pooled_h, const std::size_t& spatial_bin_x, const std::size_t& spatial_bin_y, const std::string& mode, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskQuantize(const std::vector< Data::TensorIterator >& inputs, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskReshape(const std::vector< Data::TensorIterator >& inputs, const Shape& shape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskRegionYolo(const std::vector< Data::TensorIterator >& inputs, const unsigned& coords, const unsigned& classes, const bool& do_softmax, const unsigned& num = 0, const std::vector<unsigned>& mask = {}, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskReorgYolo(const std::vector< Data::TensorIterator >& inputs, const unsigned& stride, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskNormalize(const std::vector< Data::TensorIterator >& inputs, const double& eps, const unsigned& across_spatial = 0, const unsigned& channel_shared = 0, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskPermute(const std::vector< Data::TensorIterator >& inputs, const Order& order, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskEltwise(const std::vector< Data::TensorIterator >& inputs, const std::string& eltwiseType, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskInterp(const std::vector< Data::TensorIterator >& inputs, const double& factor, const unsigned& pad_beg, const unsigned& pad_end, const unsigned& height = 0, const unsigned& width = 0, const bool& align_corners = true, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskDetectionOutput(const std::vector< Data::TensorIterator >& inputs, const int64_t& num_classes, const int64_t& keep_top_k, const double& nms_threshold, const int64_t& background_label_id, const int64_t& top_k, const bool& variance_encoded_in_target, const std::string& code_type, const bool& share_location, const double& confidence_threshold, const bool& clip_before_nms, const bool& clip_after_nms, const int64_t& decrease_label_id, const bool& normalized, const int64_t& input_height, const int64_t& input_width, const double& objectness_score, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskPriorbox(const std::vector< Data::TensorIterator >& inputs, const unsigned& flip, const unsigned& clip, const double& step_w, const double& step_h, const double& offset, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskArgmax(const std::vector< Data::TensorIterator >& inputs, const int64_t& out_max_val, const int64_t& top_k, const int64_t& axis = 99, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskTopK(const std::vector< Data::TensorIterator >& inputs, const std::string& sort, const std::string& mode, const int64_t& top_k, const int64_t& axis = 99, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskNorm(const std::vector< Data::TensorIterator >& inputs, const double& alpha, const double& beta, const std::string& region, const unsigned& local_size, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskResample(const std::vector< Data::TensorIterator >& inputs, const std::string& interpolation, const bool& antialias, const Shape& output_shape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskFakeQuantize(const std::vector< Data::TensorIterator >& inputs, const unsigned& levels, const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
mv::Data::TensorIterator uPATaskCustom(const std::vector< Data::TensorIterator >& inputs, const std::vector<uint8_t>& kernelData, const std::vector<uint8_t>& paramData, const Order& outOrder, const Shape& outShape, const DType& dType = mv::DType("Default"), const mv::QuantizationParams& quantParams = {{},{},{},{}}, const std::string& name = "");
        mv::Data::TensorIterator weightsTable(const std::vector<int64_t>& data, const Shape& shape, const DType& dType, const Order& order, const mv::QuantizationParams& quantParams = {{},{},{},{},{},{}}, const std::string& name = "");

        Data::OpListIterator getSourceOp(Data::TensorIterator tensor) override;
        void addAttr(Data::OpListIterator op, const std::string& name, const Attribute& attr) override;
        bool isValid() const override;
        bool isValid(Data::TensorIterator tensor) const override;
        bool isValid(Data::OpListIterator op) const override;
        std::string getName() const override;

    };

}

#endif //MV_OP_MODEL_HPP_
