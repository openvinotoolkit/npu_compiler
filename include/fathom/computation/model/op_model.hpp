#ifndef OP_MODEL_HPP_
#define OP_MODEL_HPP_

#include "include/fathom/api/compositional_model.hpp"
#include "include/fathom/computation/model/model.hpp"
#include "include/fathom/computation/op/ops_headers.hpp"

namespace mv
{

    class OpModel : public ComputationModel, public CompositionalModel
    {

        bool defaultControlFlow_(Data::OpListIterator op);
        bool defaultStage_(Data::OpListIterator op);
        Data::OpListIterator checkInputTensor_(Data::TensorIterator inputTensor);
        bool defineInputFlow_(Data::TensorIterator inputTensor, byte_type inputIdx, Data::OpListIterator op);

        Data::TensorIterator defineOp_(computation_graph::first_graph::node_list_iterator& opNode, Data::TensorIterator* inputs, byte_type numInputs);

    public:

        OpModel(Logger::VerboseLevel verboseLevel = Logger::VerboseLevel::VerboseWarning, bool logTime = false);
        OpModel(const ComputationModel& model);

        Data::OpListIterator switchContext(Control::OpListIterator& other);

        Data::OpListIterator getInput();
        Data::OpListIterator getOutput();
        Data::OpListIterator opEnd();

        Data::TensorIterator input(const Shape& shape, DType dType, Order order, const string& name = "");
        Data::TensorIterator output(Data::TensorIterator input, const string& name = "");
        Data::TensorIterator constant(float_type *data, size_type size, const Shape& shape, DType dType, Order order, const string& name = "");
        Data::TensorIterator constant(const dynamic_vector<float_type>& data, const Shape& shape, DType dType, Order order, const string& name = "");
        Data::TensorIterator conv2D(Data::TensorIterator input, Data::TensorIterator filters, UnsignedVector2D stride, UnsignedVector4D padding, const string& name = "");
        Data::TensorIterator fullyConnected(Data::TensorIterator input, Data::TensorIterator weights, const string& name = "");
        Data::TensorIterator maxpool2D(Data::TensorIterator input, UnsignedVector2D kernelSize, UnsignedVector2D stride, UnsignedVector4D padding, const string& name = "");
        Data::TensorIterator avgpool2D(Data::TensorIterator input, UnsignedVector2D kernelSize, UnsignedVector2D stride, UnsignedVector4D padding, const string& name = "");
        Data::TensorIterator concat(Data::TensorIterator input0, Data::TensorIterator input1, const string& name = "");
        Data::TensorIterator batchNorm(Data::TensorIterator input, Data::TensorIterator mean, Data::TensorIterator variance, Data::TensorIterator offset, Data::TensorIterator scale, float_type varianceEps, const string& name = "");
        Data::TensorIterator scale(Data::TensorIterator input, Data::TensorIterator scale, const string& name = "");
        Data::TensorIterator relu(Data::TensorIterator input, const string& name = "");
        Data::TensorIterator softmax(Data::TensorIterator input, const string& name = "");
        Data::TensorIterator add(Data::TensorIterator input0, Data::TensorIterator input1, const string& name = "");
        Data::TensorIterator subtract(Data::TensorIterator input0, Data::TensorIterator input1, const string& name = "");
        Data::TensorIterator multiply(Data::TensorIterator input0, Data::TensorIterator input1, const string& name = "");
        Data::TensorIterator divide(Data::TensorIterator input0, Data::TensorIterator input1, const string& name = "");
        Data::TensorIterator reshape(Data::TensorIterator input, const Shape& shape, const string& name = "");

        Data::OpListIterator getSourceOp(Data::TensorIterator tensor);
        bool addAttr(Data::OpListIterator op, const string& name, const Attribute& attr);
        bool isValid() const;

        GroupContext::MemberIterator addGroupElement(Data::OpListIterator element, GroupContext::GroupIterator group);
        bool removeGroupElement(Data::OpListIterator element, GroupContext::GroupIterator group);
        using ComputationModel::addGroupElement;
        using ComputationModel::removeGroupElement;

        dynamic_vector<Shape> getInputShapes(Data::OpListIterator& op);
        dynamic_vector<Shape> getOutputShapes(Data::OpListIterator& op);
       
    };

}

#endif // OP_MODEL_HPP_