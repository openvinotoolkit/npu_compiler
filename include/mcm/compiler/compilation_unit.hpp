#ifndef MV_COMPILATION_UNIT_HPP_
#define MV_COMPILATION_UNIT_HPP_

#include "include/mcm/computation/model/control_model.hpp"
#include "include/mcm/computation/model/data_model.hpp"
#include "include/mcm/op_model.hpp"
#include "include/mcm/logger/log_sender.hpp"
#include "include/mcm/pass/pass_manager.hpp"
#include "include/mcm/utils/env_loader.hpp"
#include "include/mcm/compiler/compilation_descriptor.hpp"
#include "include/mcm/compiler/compilation_profiler.hpp"
#include "include/mcm/target/kmb/runtime_model/runtime_model.hpp"

namespace mv
{

    class CompilationUnit : public LogSender
    {

        static const std::string ma2490DefTargetDescPath_;

        static const std::string ma2490DefCompDescPath_;
        static const std::string compositionalModelRecordingsPath_;

        static Logger& logger_;

        OpModel* model_;
        PassManager passManager_;
        TargetDescriptor targetDescriptor_;
        CompilationDescriptor compDescriptor_;
        const static unsigned jsonParserBufferLength_ = 256;

    public:
        std::shared_ptr<std::vector<char> > getBlob() const;

        CompilationUnit(const std::string& modelName);
        ~CompilationUnit();

        bool loadTargetDescriptor(const std::string& path);
        bool loadTargetDescriptor(Target target);
        bool loadCompilationDescriptor(const std::string& path);
        bool loadCompilationDescriptor(Target target);

        CompilationDescriptor& compilationDescriptor();
        OpModel& model();

        bool initialize();
        Element runStep();
        Element run();
        bool completed() const;

        virtual std::string getLogID() const override;

    };

}

#endif // MV_COMPILATION_UNIT_HPP_
