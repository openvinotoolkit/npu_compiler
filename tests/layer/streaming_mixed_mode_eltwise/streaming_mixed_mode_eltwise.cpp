
// The file was generated by RecordedOpModel

#include <limits>
#include <include/mcm/op_model.hpp>
#include "include/mcm/compiler/compilation_unit.hpp"

int main()
{
    using namespace mv;

    static const auto inf = std::numeric_limits<double>::infinity();

    mv::CompilationUnit unit("Mixed_mode");
    mv::OpModel& model = unit.model();
    std::string compDescPath = mv::utils::projectRootPath() + "/config/compilation/release_kmb.json";
    unit.loadCompilationDescriptor(compDescPath);
    unit.loadTargetDescriptor(mv::Target::ma2490);
    //input full of 0.5
    const auto data_0 = model.input({64, 32, 512, 1}, mv::DType("UInt8"), mv::Order("NHWC"), {{0},{0.00196078431372549},{-inf},{inf},{0},{1}}, "data");

    std::vector<int64_t> weightsData0 = mv::utils::generateSequence<int64_t> (8192, 255, 0);
//    std::vector<int64_t> weightsData0 = mv::utils::generateSequence<int64_t> (73728, 255, 0);
    //weights full of 0.1
    auto weights0 = model.constantInt(weightsData0,{1,1,512,16}, mv::DType("UInt8"), mv::Order("NCHW"), {{0},{0.0003921568627450981},{},{}});
    auto conv0 = model.conv(data_0, weights0, {1, 1}, {0, 0, 0, 0}, 1, 1,  mv::DType("UInt8"),{{0},{1},{-inf},{inf},{0},{1}} , "conv");

    model.output(conv0, mv::DType("Float16"), {{},{},{},{}});
    unit.initialize();
    unit.run();
}
