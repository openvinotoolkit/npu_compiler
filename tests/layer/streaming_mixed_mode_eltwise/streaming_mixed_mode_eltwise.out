version https://git-lfs.github.com/spec/v1
oid sha256:e57972efcc26dc6c4c89facbfb24f589450645b949352ca1962f320206326ee9
size 123008
